api/src/main/java/org/apache/iceberg/metrics/DefaultCounter.java

```diff
@@ -63,7 +63,6 @@ public void increment() {
 
   @Override
   public void increment(long amount) {
-    Math.addExact(counter.longValue(), amount);
```

Comment:
Calling `longValue()` on `LongAdd` is expensive as it performs an aggregation over cells. It basically removes all the benefits of `LongAdd` that is supposed to scale well in case of contention. We call it every time the counter is being modified. I am also not sure having an overflow (which is unlikely given that we are using longs) should fail queries, for instance. These are metrics and they should add as little overhead as possible.

<img width="1508" alt="image" src="https://github.com/apache/iceberg/assets/6235869/23d90a49-b0b3-41bb-889e-a38c699f6132">

Reply:
ctx.scanMetrics().totalFileSizeInBytes().increment(dataFile.fileSizeInBytes()) is the one that had the potential. With long as datatype, it's like 8192 Petabyte data read, so doesn't seem to be an issue as of now :) 
Just curious, since it's a public class and if someone is using it, should we mention this in javadoc?

Reply:
> since it's a public class and if someone is using it, should we mention this in javadoc?

That's a good idea, will do if we decide to proceed with the change.